### PR TITLE
refactor: change `Monadic For` syntax to `forM`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/5464

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -557,7 +557,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
     }
 
     def ForYield: Rule1[ParsedAst.Expression.ForYield] = rule {
-      SP ~ keyword("for") ~ optWS ~ ForFragments ~ optWS ~ keyword("yield") ~ WS ~ Expression ~ SP ~> ParsedAst.Expression.ForYield
+      SP ~ keyword("forM") ~ optWS ~ ForFragments ~ optWS ~ keyword("yield") ~ WS ~ Expression ~ SP ~> ParsedAst.Expression.ForYield
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:560]: Let us allow both `for` and `forM` for now-- but fine to refactor the test cases.

(Later `for` will become something that automatically selects between the two.)

File: main/test/flix/Test.Exp.ForYield.flix [hunk 2]
```diff
@@ -3,42 +3,42 @@ namespace Test/Exp/ForYield {
     @test
     def testForYield01(): Bool =
         let l =
-            for (x <- 1 :: 2 :: 3 :: Nil)
+            forM (x <- 1 :: 2 :: 3 :: Nil)
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.ForYield.flix:6]: Do we also rename the file?

Comment (jaschdoc) [reply main/test/flix/Test.Exp.ForYield.flix:6]: Done in separate PR, so merging main will do the trick

Comment (magnus-madsen) [reply main/test/flix/Test.Exp.ForYield.flix:6]: But it still called ForYield.flix or you mean a separate PR is coming?

Comment (jaschdoc) [reply main/test/flix/Test.Exp.ForYield.flix:6]: I'll push the changes, but on main / master the file has been renamed

Comment (magnus-madsen) [reply main/test/flix/Test.Exp.ForYield.flix:6]: Ah okay

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**